### ISSUE TITLE
[Installation]: Issues with installing vLLM on ROCM without sudo access

### ISSUE THREAD

**Issue by prajwal1210** [vllm-project/vllm#8042]
### Your current environment

Hi!

I have been trying to install vLLM on a cluster that has AMD MI250X GPUs using the documentation provided and running into the following issue:

`ninja: error: '/opt/rocm/lib/libamdhip64.so', needed by '/full_path_here/vllm/_core_C.abi3.so', missing and no known rule to make it`

Now, there are 3 issues:

1) I do not have sudo access on the cluster so I cannot copy libamdhip64.so. Is there a way to download the patch in some other path and have the installation use it?

2) The cluster I am on has ROCM 6.2 but going by this issue: https://github.com/vllm-project/vllm/issues/8004, it still requires this file

3) The rocm path on the cluster is not /opt/rocm but opt/rocm-6.2. I don't see a way to specify explicitly the ROCM Path to be used during the installation.

Let me know if there is a way to bypass these issues. 

### How you are installing vllm

```
pip install --upgrade pip

# Install PyTorch
pip uninstall torch -y
pip install --no-cache-dir --pre torch==2.5.0.dev20240726 --index-url https://download.pytorch.org/whl/nightly/rocm6.1

# Build & install AMD SMI
pip install /opt/rocm/share/amd_smi

# Install dependencies
pip install --upgrade numba scipy huggingface-hub[cli]
pip install "numpy<2"
pip install -r requirements-rocm.txt

# Apply the patch to ROCM 6.1 (requires root permission)
wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6 -P /opt/rocm/lib
rm -f "$(python3 -c 'import torch; print(torch.__path__[0])')"/lib/libamdhip64.so*

# Build vLLM for MI210/MI250/MI300.
export PYTORCH_ROCM_ARCH="gfx90a;gfx942"
python3 setup.py develop
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by youkaichao** [comment]
cc @hongxiayang

**Comment by hongxiayang** [comment]
Are you able to create a symbolic link to point /opt/rocm to /opt/rocm6.2?

If you are using ROCm 6.2, you can skip or ignore the following steps related to ROCm 6.1 patch: 

```
# Apply the patch to ROCM 6.1 (requires root permission)
wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6 -P /opt/rocm/lib
rm -f "$(python3 -c 'import torch; print(torch.__path__[0])')"/lib/libamdhip64.so*
```

**Comment by prajwal1210** [comment]
Hi! No, I cannot create the symlink since I don't have permissions to write to the /opt folder

**Comment by hongxiayang** [comment]
Is that possible you ask your system admin for help to check installation method or creating symlink? if ROCm is installed using `amdgpu-install -y --usecase=rocm,dkms` then symbolic links should be generated by default

**Comment by gounley** [comment]
At least for our use case, the symlink option isn't workable because we're installing to `/opt/rocm6.2.0` specifically because multiple versions of rocm are installed and actively used on the cluster.

**Comment by prajwal1210** [comment]
Same for my usecase as well. Multiple versions of rocm are installed and actively used

**Comment by hongxiayang** [comment]
Can you build your vllm using docker build method with Dockerfile.rocm? The base docker provides the /opt/rocm folder in its image.

**Comment by gounley** [comment]
Are we still restricted to rocm/6.1.x in the Dockerfile.rocm or do you expect that a base image with 6.2 will work? (particularly interested in the new stuff in https://rocm.docs.amd.com/en/latest/about/release-notes.html#improved-vllm-support)

**Comment by hongxiayang** [comment]
The Dockerfile.rocm has its own default rocm version. For your environment which is MI250x based, I think the current 6.1.x in the Dockerfile.rocm should serve your purpose.  Otherwise, you should check [ROCm/vllm](https://github.com/ROCm/vllm) if you use MI300x.

**Comment by prajwal1210** [comment]
What I wanted to know is that where is the `/opt/rocm' path hard-coded in the vllm repo (or one of it dependencies). Should it not be possible to just change the path there and make it work.

I will try the docker solution but the cluster I am working on has apptainer instead of docker and it might have limited capabilities.

**Comment by gounley** [comment]
@prajwal1210 Fairly sure we are working on the same machine; will ping you if we get something working.

**Comment by hongxiayang** [comment]
> What I wanted to know is that where is the `/opt/rocm` path hard-coded in the vllm repo (or one of it dependencies). Should it not be possible to just change the path there and make it work.
> 
> I will try the docker solution but the cluster I am working on has apptainer instead of docker and it might have limited capabilities.

Unfortunately I don't have an environment to reproduce your issue. If you can send out all the steps and logs, we can check where that /opt/rocm comes from. You can use the pastebin to share the logs.

**Comment by hongxiayang** [comment]
fyi: I just did a docker build on one mi210 machine (with rocm 6.2 installed) using the current main branch, with the following command: 
```
DOCKER_BUILDKIT=1 docker build -f Dockerfile.rocm -t vllm-rocm .
```

And it built successfully. Then you can docker run `vllm-rocm`  to get to the container, and you will find it is 6.1 inside the container as below:
```
 ls -l /opt
drwxr-xr-x 5 root    root    4096 Jun  5 17:53 amdgpu
drwxr-xr-x 1 root    root    4096 Jun  5 18:49 cache
drwxr-xr-x 1 jenkins jenkins 4096 Jun  5 23:51 conda
drwxr-xr-x 7 root    root    4096 Jun  5 23:57 mellanox
drwxr-xr-x 4 root    root    4096 Jun  6 00:00 mpi_install
drwxr-xr-x 7 root    root    4096 Jun  6 00:11 ompi
lrwxrwxrwx 1 root    root      22 Jun  5 17:57 rocm -> /etc/alternatives/rocm
drwxr-xr-x 1 root    root    4096 Jun  5 18:43 rocm-6.1.2
drwxr-xr-x 6 root    root    4096 Jun  6 00:00 ucx
```

**Comment by prajwal1210** [comment]
> > What I wanted to know is that where is the `/opt/rocm` path hard-coded in the vllm repo (or one of it dependencies). Should it not be possible to just change the path there and make it work.
> > I will try the docker solution but the cluster I am working on has apptainer instead of docker and it might have limited capabilities.
> 
> Unfortunately I don't have an environment to reproduce your issue. If you can send out all the steps and logs, we can check where that /opt/rocm comes from. You can use the pastebin to share the logs.

I get the following error:

```

-- Generating done (0.1s)
-- Build files have been written to: /lustre/orion/csc569/scratch/prajwal/vllm/build/temp.linux-x86_64-cpython-311
ninja: error: '/opt/rocm/lib/libamdhip64.so', needed by '/lustre/orion/csc569/scratch/prajwal/vllm/build/lib.linux-x86_64-cpython-311/vllm/_core_C.abi3.so', missing and no known rule to make it
```

I have the required rocm file here:
```
➜ ls /opt/rocm-6.2.0/lib/libamdhip64.so
/opt/rocm-6.2.0/lib/libamdhip64.so
```

Doing a grep on the vllm repo does not show me anywhere this path - /opt/rocm is required. 

The LD library path has the required rocm path
```
➜ echo $LD_LIBRARY_PATH
/sw/frontier/spack-envs/base/opt/linux-sles15-x86_64/gcc-7.5.0/llvm-12.0.1-izizukzg4567koqlyog5ln3xwvquuyhm/lib:/opt/rocm-6.2.0/lib/roctracer:/opt/rocm-6.2.0/lib/rocprofiler:/opt/rocm-6.2.0/lib:/opt/cray/pe/papi/7.0.1.2/lib64:/opt/cray/lib64
```

**Comment by hongxiayang** [comment]
Several things we need to check to see how we can work around your problem:

(1) Does this file exist? /lustre/orion/csc569/scratch/prajwal/vllm/build/lib.linux-x86_64-cpython-311/vllm/_core_C.abi3.so'
(2) If yes, can you do a `ldd _core_C.abi3.so` in that directory and send out the output
(3) Can you check where torch is in? `pip show torch` and send out the output.

**Comment by prajwal1210** [comment]
> Several things we need to check to see how we can work around your problem:
> 
> (1) Does this file exist? /lustre/orion/csc569/scratch/prajwal/vllm/build/lib.linux-x86_64-cpython-311/vllm/_core_C.abi3.so' (2) If yes, can you do a `ldd _core_C.abi3.so` in that directory and send out the output (3) Can you check where torch is in? `pip show torch` and send out the output.

(1) The file does not exist
(3) Output of pip show torch:

```
Name: torch
Version: 2.5.0.dev20240726+rocm6.1
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /lustre/orion/csc569/scratch/prajwal/venvs/vllm/lib/python3.11/site-packages
Requires: filelock, fsspec, jinja2, networkx, pytorch-triton-rocm, sympy, typing-extensions
Required-by: accelerate, peft, tensorizer, torchaudio, torchvision
```

**Comment by prajwal1210** [comment]
I also checked the build files in  `/build/temp.linux-x86_64-cpython-310/CMakeFiles/_core_C.dir` and found this in the build.make file:

```
# External object files for target _core_C
_core_C_EXTERNAL_OBJECTS =

/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: CMakeFiles/_core_C.dir/csrc/core/torch_bindings.cpp.o
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: CMakeFiles/_core_C.dir/build.make
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libamdhip64.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /lustre/orion/scratch/prajwal/csc569/anaconda3/envs/vllm/lib/python3.10/site-packages/torch/lib/libtorch.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /lustre/orion/scratch/prajwal/csc569/anaconda3/envs/vllm/lib/python3.10/site-packages/torch/lib/libc10.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /lustre/orion/scratch/prajwal/csc569/anaconda3/envs/vllm/lib/python3.10/site-packages/torch/lib/libc10_hip.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /lustre/orion/scratch/prajwal/csc569/anaconda3/envs/vllm/lib/python3.10/site-packages/torch/lib/libc10.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm/lib/libamdhip64.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libMIOpen.so.1.0.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm/lib/libhiprtc.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm/lib/libroctx64.so
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhipblaslt.so.0.8.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhipblas.so.2.2.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhipfft.so.0.1.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhiprand.so.1.1.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhipsparse.so.1.1.0.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libhipsolver.so.0.2.60200
/lustre/orion/csc569/scratch/prajwal/vllm-latest/build/lib.linux-x86_64-cpython-310/vllm/_core_C.abi3.so: /opt/rocm-6.2.0/lib/libamdhip64.so.6.2.60200
```

**Comment by hongxiayang** [comment]
Ok, here is what I would like you to try to use good libamdhip files for the vllm installation:
## go to your torch installation directory
```
cd /lustre/orion/csc569/scratch/prajwal/venvs/vllm/lib/python3.11/site-packages/torch/lib
```
## save a copy of old libamdhip64.so file
```
cp libamdhip64.so libamdhip64.so.sv
```
## copy the rocm6.2 files over to your torch installation 
```
cp /opt/rocm6.2.0/lib/libamdhip64.so .
cp /opt/rocm6.2.0/lib/libamdhip64.so.6 .
```
##alternatively, wget the amdhip64.so.6 from below and put it in the above torch/lib directory
```
wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6 
cp libamdhip64.so.6 libamdhip64.so
```


Now try to rebuild vllm, and see what happens.

**Comment by gounley** [comment]
This suggestion is failing for me. I'm copying the two files from `/opt/rocm-6.2.0/lib` to the necesssary directories in `lib/python3.11/site-packages/torch/lib` but I'm not getting a clean import of torch afterward:

> (base) [gounley1@login13.frontier gounley1]$ python -c "import torch"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/sw/summit/mde/med116_mde/gounley1/condavllm620/lib/python3.11/site-packages/torch/__init__.py", line 368, in <module>
    from torch._C import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^^^^
ImportError: /autofs/nccs-svm1_sw/summit/mde/med116_mde/gounley1/condavllm620/bin/../lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /opt/rocm-6.2.0/lib/libamd_comgr.so.2)

Torch is being installed via a nightly wheel, not source, if that makes a difference.

**Comment by hongxiayang** [comment]
How about you replace  using the wget method?
in your torch/lib directory:

wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6 
cp libamdhip64.so.6 libamdhip64.so

**Comment by gounley** [comment]
No luck with that either, unfortunately, using `rocm/6.1.3` or `rocm/6.2.0`.

**Comment by hongxiayang** [comment]
what is the error after you did 

in your torch/lib directory (pip show torch):

```
wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6
cp libamdhip64.so.6 libamdhip64.so
```
and then import torch.

**Comment by gounley** [comment]
`pip show torch`:

> (base) [gounley1@login07.frontier vllm]$ pip show torch
Name: torch
Version: 2.5.0.dev20240726+rocm6.1
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3
Location: /sw/summit/mde/med116_mde/gounley1/condasine613/lib/python3.11/site-packages
Requires: filelock, fsspec, jinja2, networkx, pytorch-triton-rocm, sympy, typing-extensions
Required-by:

Error message: 
> (base) [gounley1@login07.frontier vllm]$ python -c "import torch"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/sw/summit/mde/med116_mde/gounley1/condasine613/lib/python3.11/site-packages/torch/__init__.py", line 365, in <module>
    from torch._C import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^^^^
ImportError: /autofs/nccs-svm1_sw/summit/mde/med116_mde/gounley1/condasine613/bin/../lib/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /opt/rocm-6.1.3/lib/libamd_comgr.so.2)

Error message isn't present before I bring in `libamdhip64.so.6` and copy to `libamdhip64.so`

**Comment by hongxiayang** [comment]
For this issue,  /opt/rocm-6.1.3/lib/libamd_comgr.so.2 requires GLIBCXX_3.4.30 and your /bin/../lib/libstdc++.so.6 does not have it.
Here is a work-around to try.
Go to your torch/lib directory:
```
ldd libamdhip64.so.6
```
Anything (.so) that points to your /opt/rocm-6x directory, you just do a copy from your torch/lib directory using the one without version number.

For example:
```
cp libamd_comgr.so libamd_comgr.so.2
```

Please let me know whether this solves the GLIBCXX version issue.

**Comment by hongxiayang** [comment]
Please let me know whether the above work-around steps help. This will help us to work on an improvement/fix to address this situation in the future.

**Comment by gounley** [comment]
> (base) [gounley1@login03.frontier lib]$ ldd libamdhip64.so.6
ldd: warning: you do not have execution permission for `./libamdhip64.so.6'
	linux-vdso.so.1 (0x00007f13ba703000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f13b9171000)
	librocprofiler-register.so.0 => /opt/rocm-6.1.3/lib/librocprofiler-register.so.0 (0x00007f13b90d3000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f13b90c9000)
	libamd_comgr.so.2 => /opt/rocm-6.1.3/lib/libamd_comgr.so.2 (0x00007f13b0d79000)
	libhsa-runtime64.so.1 => /opt/rocm-6.1.3/lib/libhsa-runtime64.so.1 (0x00007f13b0a9e000)
	libnuma.so.1 => /usr/lib64/libnuma.so.1 (0x00007f13b0800000)
	libstdc++.so.6 => /usr/lib64/libstdc++.so.6 (0x00007f13b05bb000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f13b046f000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f13b0a7a000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f13b0a56000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f13b0278000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f13ba705000)
	libz.so.1 => /usr/lib64/libz.so.1 (0x00007f13b0a3b000)
	libzstd.so.1 => /usr/lib64/libzstd.so.1 (0x00007f13b0147000)
	libtinfo.so.6 => /lib64/libtinfo.so.6 (0x00007f13b0a0c000)
	libelf.so.1 => /usr/lib64/libelf.so.1 (0x00007f13afe00000)
	libdrm.so.2 => /opt/amdgpu/lib64/libdrm.so.2 (0x00007f13b0130000)
	libdrm_amdgpu.so.1 => /opt/amdgpu/lib64/libdrm_amdgpu.so.1 (0x00007f13b0120000)
	
Within `torch/lib`, I copied `libamd_comgr.so` to `libamd_comgr.so.2`, `libhsa-runtime64.so` to `libhsa-runtime64.so.1`, `libdrm.so` to `libdrm.so.2`, and `libdrm_amdgpu.so` to `libdrm_amdgpu.so.1`. However, I'm still seeing the same error. 

Maybe I am misunderstanding - should I be copying these files from `/opt/rocm-6.1.3`?

**Comment by hongxiayang** [comment]
pytorch 2.5 with rocm 6.2 wheel is now available for test. If your system is ROCm 6.2, please use this wheel to install pytorch. This should work with your rocm 6.2:
```
python3 -m pip install --no-cache-dir --pre \
                torch==2.5.0 torchvision  --index-url https://download.pytorch.org/whl/test/rocm6.2
```

**Comment by prajwal1210** [comment]
I tried the new build instructions with Pytorch 2.5 and RocM 6.2. I still get this error:
```
gmake[3]: *** No rule to make target '/opt/rocm/lib/libamdhip64.so', needed by '_core_C.abi3.so'.  Stop.
gmake[3]: *** Waiting for unfinished jobs....
```

I think somehow Cmake is finding /opt/rocm as a default path to look into. Even though it is not hard-coded anywhere.

**Comment by prajwal1210** [comment]
> > (base) [[gounley1@login03.frontier](mailto:gounley1@login03.frontier) lib]$ ldd libamdhip64.so.6
> > ldd: warning: you do not have execution permission for `./libamdhip64.so.6'
> > linux-vdso.so.1 (0x00007f13ba703000)
> > libdl.so.2 => /lib64/libdl.so.2 (0x00007f13b9171000)
> > librocprofiler-register.so.0 => /opt/rocm-6.1.3/lib/librocprofiler-register.so.0 (0x00007f13b90d3000)
> > librt.so.1 => /lib64/librt.so.1 (0x00007f13b90c9000)
> > libamd_comgr.so.2 => /opt/rocm-6.1.3/lib/libamd_comgr.so.2 (0x00007f13b0d79000)
> > libhsa-runtime64.so.1 => /opt/rocm-6.1.3/lib/libhsa-runtime64.so.1 (0x00007f13b0a9e000)
> > libnuma.so.1 => /usr/lib64/libnuma.so.1 (0x00007f13b0800000)
> > libstdc++.so.6 => /usr/lib64/libstdc++.so.6 (0x00007f13b05bb000)
> > libm.so.6 => /lib64/libm.so.6 (0x00007f13b046f000)
> > libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f13b0a7a000)
> > libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f13b0a56000)
> > libc.so.6 => /lib64/libc.so.6 (0x00007f13b0278000)
> > /lib64/ld-linux-x86-64.so.2 (0x00007f13ba705000)
> > libz.so.1 => /usr/lib64/libz.so.1 (0x00007f13b0a3b000)
> > libzstd.so.1 => /usr/lib64/libzstd.so.1 (0x00007f13b0147000)
> > libtinfo.so.6 => /lib64/libtinfo.so.6 (0x00007f13b0a0c000)
> > libelf.so.1 => /usr/lib64/libelf.so.1 (0x00007f13afe00000)
> > libdrm.so.2 => /opt/amdgpu/lib64/libdrm.so.2 (0x00007f13b0130000)
> > libdrm_amdgpu.so.1 => /opt/amdgpu/lib64/libdrm_amdgpu.so.1 (0x00007f13b0120000)
> 
> Within `torch/lib`, I copied `libamd_comgr.so` to `libamd_comgr.so.2`, `libhsa-runtime64.so` to `libhsa-runtime64.so.1`, `libdrm.so` to `libdrm.so.2`, and `libdrm_amdgpu.so` to `libdrm_amdgpu.so.1`. However, I'm still seeing the same error.
> 
> Maybe I am misunderstanding - should I be copying these files from `/opt/rocm-6.1.3`?

I also tried this with the latest pytorch and I get the same error after copying the files to torch/lib

**Comment by mattaezell** [comment]
Is this the same issue as:
https://github.com/pytorch/pytorch/issues/131701
https://github.com/pytorch/pytorch/pull/136283

**Comment by prajwal1210** [comment]
Looking at the issue, I think this could be it. The pytorch wheel requires /opt/rocm/lib to be the path for libamdhip64.so and so it does not work when building vLLM from source. I guess I will just have to wait for the issue to be resolved in Pytorch to be able to try it out with vLLM. Will try building torch from source according to the issue mentioned. Thanks!

**Comment by hongxiayang** [comment]
The pytorch fix for hard-coded rocm path is available in pytorch nightly now.

You don't need to build pytorch by yourself. If your rocm is 6.1, you can install pytorch with:
```
pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm6.1
```

Otherwise, if your rocm is 6.2, you will install pytorch as:
```
pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm6.2
```

**Comment by gounley** [comment]
Belatedly reporting back - installation works as expected now.

**Comment by hongxiayang** [comment]
closing the issue as the issue is resolved
